app/controllers/users_controller.rb

```diff
@@ -23,7 +24,10 @@ def new
   def create
     user = User.new(user_params)
 
-    authenticate!(:password) if user.save
+    if user.save!
```

Comment:
don't let user.save! here as you'll get an exception for validation errors. You can put an else if you want to show an error when trying to sign up. That would be much better probably.